rfcs/003-randomness.md

```diff
@@ -0,0 +1,151 @@
+determining the winning block a non-deterministic operation. As such, as long
+as there is at least one honest collator that is able to submit a winning block
+every so often, a dishonest SP will always be detected.
+
+Furthermore, as long as we use the latest block hash available (i.e. the
+parent block of the one being produced) even if the SP can bias the resulting
+hash, it will only provide them with an advantage of 6 seconds (block production
+time) + challenge lookback (10 minutes for the production testnet); additionaly
+this advantage does not stack, meaning that being 10 (or any arbitrary amount)
+minutes early to a proof does not enable the SP to start the next proof 10 (or
+any arbitrary amount) minutes early.
+
+### BABE Randomness
+
+As part of BABE consensus, VRFs are used to select the validator that will
+propose a block — the VRF [randomness is exposed in the relay chain proof][1],
+meaning we just need to "pull" that randomness into our parachain (easier
+said than done); this is the approach taken by Moonbeam ([MB1][2], [MB2][3]),
+and StorageHub ([SH1][4], [SH2][5]).
+
+This approach involves injecting the BABE randomness value into parachain
+storage, then, when necessary fetch it for usage. The main advantage is that
+this approach is verifiable, the main disadvantage is that the randomness
+takes a long time to refresh as it is dependent on BABE epochs (which last
+4 hours on mainnet).
+
+### Local-VRF
+
+The Local-VRF is another Moonbeam solution to the randomness problem, this
+approach is simpler, in that the collator producing the block simply generates
+and stores a VRF output on-chain.
+
+This approach suffers the same issue that the block hash suffers — it is
+biasable, e.g. the collator may generate multiple values until one that benefits
+them comes up. It may be possible to mitigate this by using the previous VRF
+value or block hash as input to the VRF — this limits malicious collators since
+they are no longer able to "roll the dice" multiple times until a suitable value
+emerges.
+
+This approach also carries the issue that "non-use-case-specific" VRF
+implementations are not common, while VRFs are in production use across
+blockchains, however none of them are "plug and play".
+
+### drand Pallet
+
+The drand network is the randomness source of multiple blockchain networks,
+however, Polkadot is not one of them. Building a bridge to drand is not a
+trivial task, but some folks have started building just that. The
+[drand pallet][6] provides randomness from drand's Quicknet while being
+pallet-randomness compliant.
+
+Validation of the pulses can be done as part of the proof of validity of the
+parachain block, as such, if some collator submits a "fake" pulse, the validator
+should be able to detect and punish the collator accordingly.
+
+The main issue with the crate is that it is not ready for production use.
+
+## Conclusion
+
+This document covers multiple randomness methods, however, none of them
```

Comment:
I think, RFC should select one of the solution (according to the author), as this records a decision that have been made.

Plus, I think a table comparing those approaches would help.